### ISSUE TITLE
Apple will not allow QGIS on macOS to open / unverified developer

### ISSUE THREAD

**Issue by wnrand** [qgis/QGIS#55663]
### What is the bug or the crash?

I have been going around on this for several weeks now, trying various previous suggestions to get MacOS X to allow me to open QGIS. I have talked to our IT department and Apple and both keep referring me back to QGIS to solve this issue.

Control-click to select "Open" does not work (even for the first time opening with a fresh install.) Clicking "Open anyway" does not work. Rebooting into safe mode does not fix the issue. I believe I tried the hint to use a terminal script and that did not work either. Is there any other option to get QGIS to launch on Mac?

![Screen Shot 2023-11-28 at 3 23 58 PM](https://github.com/qgis/QGIS/assets/29783995/dbd5d6b7-dfde-4c91-8b6b-14bd316d7bf3)

### Steps to reproduce the issue

Navigate to QGIS application
Control-click to bring up context menu, select "Open"
Error message pops up that says "“QGIS” cannot be opened because the developer cannot be verified."
Navigate to System preferences, open "Security & Privacy"
Click "Open Anyway" button on "General" tab
Back to error message

### Versions

QGIS 3.34.1-Prizren

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

I fully deleted all QGIS supporting files before doing a fresh installation with no change in behavior

**Comment by agiudiceandrea** [comment]
@wnrand, have you followed the instruction displayed on the [Download](https://qgis.org/en/site/forusers/download.html#mac) page?

> To enable QGIS, control-click on its icon in your Applications folder and select Open in the context menu. A confirmation dialog will display where you need to click the Open button again.

![image](https://github.com/qgis/QGIS/assets/16253859/3591dea1-238f-4eb9-8097-8e44fff9b8ea)

**Comment by agiudiceandrea** [comment]
@wnrand, could you please specify if you have followed the instruction displayed on the [Download](https://qgis.org/en/site/forusers/download.html#mac) page?

> After installing QGIS, the first launch attempt may fail due to security protections. To enable QGIS, control-click on its icon in your Applications folder and select Open in the context menu. A confirmation dialog will display where you need to click the Open button again. This only has to be done once.

![image](https://github.com/qgis/QGIS/assets/16253859/3591dea1-238f-4eb9-8097-8e44fff9b8ea)

Do you have issue installing QGIS or launching QGIS after installation?

Please also specify what is the exact version of your OS.

**Comment by pathmapper** [comment]
Here are some infos from Apple: https://support.apple.com/HT202491#openanyway

**Comment by jackvaughanjr** [comment]
> @wnrand, could you please specify if you have followed the instruction displayed on the [Download](https://qgis.org/en/site/forusers/download.html#mac) page?

I have tried the instructions using both the LTS and most recent versions.

> Does the issue occur trying to install QGIS or trying to launch QGIS after installation?

This occurs when trying to launch QGIS after installation.

> Please also specify what is the exact version of your OS and how did you download the QGIS installer.

Currently running Sonoma 14.1.2 and I downloaded directly from the site.

**Comment by agiudiceandrea** [comment]
@jackvaughanjr, thanks. It would be very useful if you indicated in detail all the steps followed and what went wrong and the exact error messages, in respect to the available instruction available on qgis.org.

**Comment by wnrand** [comment]
> @wnrand, could you please specify if you have followed the instruction displayed on the [Download](https://qgis.org/en/site/forusers/download.html#mac) page?
> 
> > After installing QGIS, the first launch attempt may fail due to security protections. To enable QGIS, control-click on its icon in your Applications folder and select Open in the context menu. A confirmation dialog will display where you need to click the Open button again. This only has to be done once.
> 
> ![image](https://private-user-images.githubusercontent.com/16253859/291725803-3591dea1-238f-4eb9-8097-8e44fff9b8ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMzYxMzIsIm5iZiI6MTcwOTMzNTgzMiwicGF0aCI6Ii8xNjI1Mzg1OS8yOTE3MjU4MDMtMzU5MWRlYTEtMjM4Zi00ZWI5LTgwOTctOGU0NGZmZjliOGVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAxVDIzMzAzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2NTZlMDEyYzRjZjJjZTFlNmVmNDkzNzRiY2MwYjlmNWMxZmJkNzVlOTJiZDdlNjJmYTEwNWNiOTczNjE1YWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6o6xvkcJE-vtgR-dpVRRqjL0MsKnVG-78wTq665jnbw)
> 
> Does the issue occur trying to install QGIS or trying to launch QGIS after installation?
> 
> Please also specify what is the exact version of your OS and how did you download the QGIS installer.

I gave up trying to get the new QGIS to work and went back to using the old version that Mac OS would allow to launch. 

I tried all those options. I deleted the old install a few times, reinstalled the newest version fresh and then -- first thing -- try and open QGIS with the control-click and select "Open". It never worked.

I am now on a new workstation and trying to get any QGIS to work and am still getting the same error when first trying to launch QGIS. No option is given to continue. System preferences/Security/General does not allow it to continue launching either

<img width="372" alt="Screen Shot 2024-03-01 at 3 20 52 PM" src="https://github.com/qgis/QGIS/assets/29783995/4392df18-6a96-4a29-a0ce-6041f3b84e2b">

**Comment by wnrand** [comment]
I had given up on trying to solve this issue and decided to just use the old version of QGIS that I had been able to install previously. But my workstation died last week and I'm now trying to get any version of QGIS to launch on the new workstation I have been given. I tried a fresh install and first action was to use the control-click to get the contextual menu and then choose "Open". But Mac OS does not give me the option to continue and the system preferences in the Security tab only trys to relaunch QGIS when I click "Open Anyway". I have found no way around this. Are there any other ideas for a way to get Mac OS to allow QGIS to launch?
![Uploading Screen Shot 2024-03-01 at 3.20.52 PM.png…]()

**Comment by chris-prener** [comment]
I ran into this problem today - I work on a macOS machine that is managed by my employer. Our profiles are set to only allow installations from identified developers. I end up in the same loop that @wnrand does - there is no way to open QGIS when this security requirement is being enforced at the enterprise level. Our IT folks told me that my two options are to get the QGIS team to participate in the Apple identified developer program or use ArcGIS. Would love to hear if this is something that could be added to your roadmap!

![image](https://github.com/qgis/QGIS/assets/13595578/86dc9a2b-7266-4756-97d5-2af7454623fb)

**Comment by DoZiBo1** [comment]
> I ran into this problem today - I work on a macOS machine that is managed by my employer. Our profiles are set to only allow installations from identified developers. I end up in the same loop that @wnrand does - there is no way to open QGIS when this security requirement is being enforced at the enterprise level. Our IT folks told me that my two options are to get the QGIS team to participate in the Apple identified developer program or use ArcGIS. Would love to hear if this is something that could be added to your roadmap!
> 
I have the exact same problem. The suggested fixes from QGIS website simply do not work.

**Comment by agiudiceandrea** [comment]
@DoZiBo1, are you using a macOS machine that is managed by your employer?

**Comment by DoZiBo1** [comment]
@agiudiceandrea Yes. Support department says it needs to be signed. No way for me to circumvent it.
QGIS [support page here](https://www.qgis.org/en/site/forusers/alldownloads.html#mac-os-x-macos) even claims to provide "signed installers" for macOS, which does not seem to be true. I can find no further reference which version is signed.

**Comment by agiudiceandrea** [comment]
@DoZiBo1, AFAIK the macOS dmg installer is signed by the "Developer ID Application: Open Source Geospatial Foundation (4F7N4UDA22)" identity, while the macOS dmg installer is not "notarized".

**Comment by DoZiBo1** [comment]
@agiudiceandrea I don't know what this means. Can you translate this into a conclusion? Does it mean, it's signed but in the wrong way/improperly, or it is not signed in all the necessary places? Or is it signed and the problem lies elsewhere?

**Comment by DoZiBo1** [comment]
@agiudiceandrea Thank you for clarification. If I see this correctly, there is already open issues regarding notarization of QGIS: [issue 218](https://github.com/qgis/QGIS-Enhancement-Proposals/issues/218) which refers to [issue 270](https://github.com/qgis/QGIS-Enhancement-Proposals/issues/270) where the necessary work is supposed to be included.

**Comment by greg-vernon** [comment]
It is not really reasonable to be asking users to install packages that go around the security measures that are built into MacOS.  At best it's sloppy, at worst a system could be potentially compromised.  There is also the case of failing a security audit if you're in a company that requires auditing. 

I worked at a CA for a number of years, and I've also got some experience with dealing with Apple dev stuff and packaging. If I can help out somehow with getting this fixed, I'd be happy to help.

**Comment by BiosPlus** [comment]
Same issue here, I have users who would like to install the application though the signing and notarisation issues are a blocker in terms of application security.

**Comment by tatjabecker** [comment]
I am not sure if it is the correct thing to do, but what helped me solve this issue was changing my security & privacy settings. In Accessibility I added QGIS to the list of apps that are allowed to control my computer. Then in the Security & Privacy settings, where it states that QGIS is blocked you can click "Open anyway". Voila it opened! Maybe the first step was not necessary, because I wasn't paying attention if the "Open anyway" option was available, nevertheless it is worth to try these two steps if you haven't already.

**Comment by greg-vernon** [comment]
> I am not sure if it is the correct thing to do, but what helped me solve this issue was changing my security & privacy settings. In Accessibility I added QGIS to the list of apps that are allowed to control my computer. Then in the Security & Privacy settings, where it states that QGIS is blocked you can click "Open anyway". Voila it opened! Maybe the first step was not necessary, because I wasn't paying attention if the "Open anyway" option was available, nevertheless it is worth to try these two steps if you haven't already.

You can definitely make things work this way.  However, from a security standpoint, it's not really the right way to be doing things.

**Comment by aileenclarke** [comment]
I'm having the same issue on an employer controlled Mac. Does anyone know what the newest version of QGIS is that doesn't cause this problem? I'm stuck on 3.16 for now.

**Comment by kentr** [comment]
Possibly related to #55930. 

I agree with others that it's unreasonable to require users to bypass the OS security features. I won't even do this on my own computer in most cases.

**Comment by Newkid76** [comment]
Hey everyone, I had the same issue as I was trying to install the latest version in my company's mac and this fixed it: 

1. Go to settings 
2. Privacy & Security 
3. Scroll down until Security 
4. You will see the blocked section called "Allow applications downloaded from": "App Store" and "App Store and identified developers" - you will not be able to do anything here 
5. But below, you will see a message about QGIS that you'll need to approve (by adding your password)

**Comment by BiosPlus** [comment]
@Newkid76, Your company Mac administrator doesn't seem to push a strict gatekeeper policy. On devices where we push a gatekeeper policy there is no work around aside from `xattr`.

**Comment by justinbb** [comment]
@greg-vernon QGIS on Mac has had some problems for a few years now. You may have noticed that complaints fall into two broad categories, lack of notarization (as here) and out-of-date components / missing Apple Silicon binaries. The two categories are unfortunately intertwined.

There was an attempt made to produce a compilant Mac QGIS bundle a few years ago: the result was [QGIS-Mac-Packager](https://github.com/qgis/QGIS-Mac-Packager/). The project was too broad in its aims, then fell far short of them. It is in two parts:
- a custom dependency manager which cannot be updated (the root cause for the numerous bug reports about out-of-date components, lack of Apple Silicon binaries, etc.)
- a bundler + installer-maker. The bundle thus produced is signed but it is not eligible for notarization.
QGIS-Mac-Packager is still used to produce the downloadable Mac installers on the QGIS site.

On dependency management / creation of a working executable:
QGIS-Mac-Packager's dependency manager part, obsolete and unmaintainable, has to be replaced.
More recently, @m-kuhn made an alternative [conda-based Mac installer](https://github.com/opengisch/qgis-conda-builder) which is functional but does not seem to have been found fully satisfactory.
The [QGIS installation guide](https://www.qgis.org/resources/installation-guide/#macports) has recommended MacPorts for some time now, which is a good basis for Mac QGIS users with the permission and ability to manage MacPorts – only a part of the user community.
There has been [a proposal](https://github.com/qgis/QGIS-Enhancement-Proposals/issues/292) to move to Microsoft vcpkg for Windows builds, a cross-platform solution which would likely give good results on Mac as well.
Any of these methods are (either theoretically or in reality) capable of getting together the code and resources to make an up-to-date, working QGIS with Apple Silicon binaries. But they still don't solve the notarization problem.

On organizing and notarizing the bundle:
Organizing the bundle so it can be notarized is a non-trivial problem, as notarization requires strict separation of code and data. The QGIS-Mac-Packager's bundling part _might_ be able to serve as the basis for a better Mac bundle (I haven't dared look). There is further discussion (including links to documentation, history, etc.) at https://github.com/qgis/QGIS-Mac-Packager/issues/160.

**Comment by greg-vernon** [comment]
@justinbb, @miceg seems to have described the problem fairly well.  For those of us who are fairly techy, and have the time to mess around with macports, that could be the answer.  Though, TBH, it's going to be a lot less work to set up a separate system to run Linux in order to comply with whatever security requirements employers require.

For those are not technically inclined, and those who have worry about security compliance, the mac is probably not going to be a QGIS platform that they can use.  None of the options supports an installation method that actually does any auditing of the packages, which is what Apple means to be doing with the notarization process.

What might be useful is to have a look at what the [PostgresApp](https://github.com/PostgresApp) project is doing.

**Comment by kentr** [comment]
I had success using QGIS inside a VM running Lubuntu with Virtualbox 6.

Virtualbox is notorized.

**Comment by greg-vernon** [comment]
@kentr That works for the more techie folks out there.  Still there are serious shortcomings.  You still have the overhead of maintaining the Virtualbox software, and you've got users now having to manage whatever Linux that's running there.  More importantly though, your video RAM is fairly limited with Virtualbox.  It's probably just easier to have a dedicated Linux system, and use it remotely via VNC or the equivalent. 

There are a lot of workarounds that work for the more technically inclined among us.  However, this isn't a good situation for the less technical users, and while there are some of us Geographers out there who are quite technical, there are many who are not. 

@justinbb macports isn't a great solution, as it tends to take over the machine.  Years ago, I was doing some mac configuration work, and I decided to get rid of macports entirely, as it doesn't play well with others. For example, if you're using macports, [it wants to own /usr/local](https://trac.macports.org/wiki/FAQ#usrlocal), and that's a major problem. 
Homebrew would be a better solution, except, as noted, it's using the bundle from qgis.org.  Still, if homebrew was used to build QGIS it could work well.  It might be the best interim solution until a notarized package can be built. Though, there could be some issues with python and certificates. 

I would say fixing the install for the mac in general is probably way more important at this point than building Apple Silicon binaries.  Of course, that's just my opinion. :)

**Comment by m-kuhn** [comment]
Thanks all for your comments.

At the current point in time, I recommend using conda (command line) to install QGIS for anyone more "techy" out there.

At the same time, I am still positive that I will find some time at some point to look into the nasty details of rpath, python, dmg packaging (and finally hopefully notarization) using a vcpkg build.

**Comment by greg-vernon** [comment]
@m-kuhn conda looks to be a **much** better solution than macports.   Thanks for all your work on the conda packaging!  

DMG packaging and the notarization with that shouldn't be too bad after you have an install properly set up.

**Comment by justinbb** [comment]
> macports isn't a great solution, as it tends to take over the machine. Years ago, I was doing some mac configuration work, and I decided to get rid of macports entirely, as it doesn't play well with others. For example, if you're using macports, [it wants to own /usr/local](https://trac.macports.org/wiki/FAQ#usrlocal), and that's a major problem. Homebrew would be a better solution, except, as noted, it's using the bundle from qgis.org. Still, if homebrew was used to build QGIS it could work well. It might be the best interim solution until a notarized package can be built. Though, there could be some issues with python and certificates.
> 
> I would say fixing the install for the mac in general is probably way more important at this point than building Apple Silicon binaries. Of course, that's just my opinion. :)

@greg-vernon I defer to @m-kuhn on the way forward for QGIS Mac builds. To reply briefly to other items: I should have made a distinction between ways of putting together a Mac build inside the QGIS project for distribution, and ways for end-users to get a Mac build these days, while waiting for the build problem to be solved. In the case of the QGIS project, homebrew was apparently used to assemble dependencies in the days before the dreaded Packager, and m-kuhn worked on conda initially with the intention (if I understand correctly) of using it as a basis for a distribution; in the end, he is surely right that vcpkg is a better tool than any of the preceding for the purpose. (If nothing else, it would give the QGIS project control over its own integration-and-build process.)

For end-users, different people have different experiences with homebrew, MacPorts, and conda, and this colors their perception. Homebrew, as you mention, is not an option unless someone completes a "formula" for QGIS. Both MacPorts and conda have their issues; whether an issue is disqualifying or not is often a matter of personal preference. (E.g. I don't find that MacPorts takes over my machine, and have used it in parallel with homebrew – but with limited usage of both. I find that conda is great in some ways and annoying in others. Etc.) And, as you also mentioned, neither of them is a good solution for less technical users.

Fixing the install for Mac is indeed way more important at this point than building Apple Silicon binaries, as QGIS compiles and builds fine on Apple Silicon (and has done so for a long time) – it's simply not an issue. The lack of an "official" Apple Silicon installer is instead a side-effect of the problem of fixing the (bundling and) install for Mac, alongside the other terrible side-effect of official Mac builds having out-of-date GDAL, PROJ, and other dependencies.

**Comment by greg-vernon** [comment]
@justinbb @m-kuhn's idea using vcpkg looks promising.  It looks like it handles libraries in a sane way, so then it should also work much better for the dependencies.

**Comment by greg-vernon** [comment]
@justinbb @m-kuhn, I was digging around in PostgresApp, and found a reference to [create-dmg](https://github.com/create-dmg/create-dmg?tab=readme-ov-file).  Perhaps it's useful?

**Comment by m-kuhn** [comment]
[`create-dmg` is also used in qfield](https://github.com/opengisch/QField/blob/master/platform/macos/CPackMacDeployQt.cmake.in#L10-L11), it's useful to create a dmg from a .app indeed.

**Comment by jlungo** [comment]
This solution here worked for me. 

System Settings --> Privacy and Security --> Scroll down and you will see the app. Click Open anyway, Enter your password. Then it works.

https://support.apple.com/en-us/102445#openanyway

**Comment by agiudiceandrea** [comment]
@justinbb, is it a different procedure than the one already described at https://qgis.org/download/#tips-for-first-launch ?

"For macOS Sequoia and newer: To enable QGIS, command-click its icon in your Applications folder and select Open from the context menu. A warning dialog will appear; click the Done button. Next, navigate to System Settings > Privacy & Security and scroll to the Security section. You should see a message stating that “QGIS” was blocked to protect your Mac. Click Open Anyway. A confirmation dialog will appear; click Open Anyway again. This only has to be done once."

**Comment by justinbb** [comment]
@agiudiceandrea No, the procedure is not different. The QGIS download page has the correct instructions, corresponding to Apple's help file https://support.apple.com/102445
The only thing I would suggest changing is to add the macOS major version numbers – some people will recognize their Mac system more easily by the number. "For macOS Sonoma (14) and earlier", "For macOS Sequoia (15) and newer".

**Comment by agiudiceandrea** [comment]
@justinbb I apologize. I would have tagged @jlungo .

**Comment by jamieodonnell** [comment]
Are there _any_ QGIS versions that do pass the Mac OS verification?

**Comment by jodygarnett** [comment]
OSGeo did purchase a certificate so that QGIS and other community applications could be signed. While we stopped making GeoServer installers this was setup for the QGIS project to use also.

**Comment by m-kuhn** [comment]
Thanks @jodygarnett I'll get to you in a few days

**Comment by agiudiceandrea** [comment]
AFAIK signing is not sufficient. A macOS app needs to be notarized, too.

**Comment by jodygarnett** [comment]
Fair enough, looks like that changed in 2023. Still notarization process seems like work in need of volunteer / funding (rather than something insermountable.)

**Comment by greg-vernon** [comment]
> Fair enough, looks like that changed in 2023. Still notarization process seems like work in need of volunteer / funding (rather than something insermountable.)

It should be relatively straightforward, and the notarization process is fairly well documented by Apple.  However, the various libraries and other packages included in building QGIS will need to be included in the notarization process, as a part of the process is auditing those parts of the build.  If those fail, so will the notarization process.

Also, this process may change this week. So it's worth paying attention to the presentations on the notarization process during this week's WWDC.

**Comment by m-kuhn** [comment]
I have worked on a first notarized version in the last weeks and have a first version. It's based on https://github.com/qgis/QGIS/pull/60039 and includes a recent master version of QGIS (built with Qt6). It's currently signed with OPENGIS.ch certificates but will be migrated to an OSGeo certificate in the future.

It can be downloaded from the bottom of this page here:
https://github.com/opengisch/qgis-notarize/actions/runs/15550334708 (Artifacts)

Feedback welcome, I am mostly interested for anything related to signing or entitlements https://developer.apple.com/documentation/bundleresources/entitlements, but I also had to change the internal file structure to make the notarization process happy, so that might also have caused some issues.
For issues that are not clearly assignable to this, please Download the [unnotarized MacOS Qt6 build](https://github.com/qgis/QGIS/actions/runs/15379869828/artifacts/3238054764) and check if it's also present there and if yes, please leave a comment there.

**Comment by greg-vernon** [comment]
It looks like the place to grab the .dmg is actually https://github.com/opengisch/qgis-notarize/actions/runs/15550334708/artifacts/3293180921

This is in the Upload Artifact step.

This looks like it installed fine on my MacBook Pro M1 Max running Sonoma 14.7.6.  Great job @m-kuhn ! :)  I can test installing it on my 2018 Intel MacBook Pro next.  That dual boots to Sonoma 14.7.6 and Sequoia.

I did have to copy & paste it into the /Applications folder.  Drag & drop simply created a shortcut/link.

**Comment by jodygarnett** [comment]
Thanks @m-kuhn that is a great service to QGSI and the foss4g community.

**Comment by m-kuhn** [comment]
Thanks for testing

@greg-vernon I absolutely want to make the installer better looking (like the one that's done in https://github.com/qgis/QGIS/pull/60039, where you can directly drag and drop it inside the same window, that's on the todo list.

**Comment by greg-vernon** [comment]
> Thanks for testing
> 
> [@greg-vernon](https://github.com/greg-vernon) I absolutely want to make the installer better looking (like the one that's done in [#60039](https://github.com/qgis/QGIS/pull/60039), where you can directly drag and drop it inside the same window, that's on the todo list.

@m-kuhn I've installed it on Sonoma (2018 MacBook Pro & 2020 iMac) & Sequoia (2018 MacBook Pro) on Intel, and Sonoma & Sequoia on my M1 MacBook Pro.  No problems with installing, and opening QGIS works when opened. This is great! :)

**Comment by m-kuhn** [comment]
Thanks for the checks @greg-vernon 
I just improved the installer and the new one looks a lot nicer:

https://github.com/opengisch/qgis-notarize/

**Comment by greg-vernon** [comment]
> Thanks for the checks [@greg-vernon](https://github.com/greg-vernon) I just improved the installer and the new one looks a lot nicer:
> 
> https://github.com/opengisch/qgis-notarize/actions/runs/15579280018

@m-kuhn the new installer is MUCH nicer! :)  Looks pretty good now.  I tested Sequoia and Sonoma with Intel, and Sequoia with M1 Max arm. Installs fine on all three.  Seems like the bug could probably be marked done now.

**Comment by kentr** [comment]
This is awesome. It’s a big benefit to the QGIS community.

**Comment by justinbb** [comment]
Thank you @m-kuhn for achieving this long-awaited goal!

For those who may have been wondering, Howard Oakley (very technical Mac blogger and author of intriguing utility software) has written a detailed explanation of why notarization has become so important. https://eclecticlight.co/2024/10/01/living-without-notarization/

I wonder also if Matthias might have time (!!) at some point to write a short article about how to get a Linux-origin open source app of great complexity, such as QGIS, into a state of organization such that it can be notarized. There is still quite a bit of FUD concerning the issue and QGIS is a great example (if QGIS can do it, just about anyone else can too!).

**Comment by rkettelerij** [comment]
Tested on Sequoia 15.5 on M3, works perfect. Thanks @m-kuhn.

**Comment by encieN** [comment]
I cannot get the latest build to work on managed computers (macOS 15.5). 

1. I installed the LTR version from https://www.qgis.org/download/ first, but couldn't get past gatekeeper policy via System Preferences. 
2. I then removed the LTR version by dragging from Finder > Apps to Trash. I then installed the notarized version. However, it keeps crashing at startup. 
3. I've tried removing the folder `~/Library/Application Support/QGIS`, rebooting and starting QGIS-master-08dd318.app again, to no avail.

Here's the crash report:
[qgis-master-crash-report.txt](https://github.com/user-attachments/files/20995950/qgis-master-crash-report.txt)

Any clues as to what may be the issue?

**Comment by christophfriedrich** [comment]
I was super excited that someone had fixed the cause just days before I ran into this issue, only to find out I have the same problem as @encieN :( My setup is very similar, company MacBook that is somehow managed by our IT department, MacOS 15.4.1, QGIS-master-b927df8. If it helps, I can also share my crash report.

For now I went with the normal LTR version and clicking "Open Anyway" here:

![Image](https://github.com/user-attachments/assets/367a62c9-f90a-4fd7-b33f-ca5daf131939)

**Comment by m-kuhn** [comment]
In the folder `/Applications/QGIS-master-[gitsha].app/Contents/PlugIns/crypto` there are a few dylibs, I wonder if removing those will "fix" it. This will disable the authentication system, but it would be interesting if this is the cause (and if yes, which of the 3 plugins it is) @encieN @christophfriedrich

**Comment by encieN** [comment]
> In the folder `/Applications/QGIS-master-[gitsha].app/Contents/PlugIns/crypto` there are a few dylibs, I wonder if removing those will "fix" it. This will disable the authentication system, but it would be interesting if this is the cause (and if yes, which of the 3 plugins it is) [@encieN](https://github.com/encieN) [@christophfriedrich](https://github.com/christophfriedrich)

Thank you! 

I've got the following dylibs in `/Applications/QGIS-master-[gitsha].app/Contents/PlugIns/crypto`:
- `libqca-gnupg.dylib`
- `libqca-logger.dylib`
- `libqca-ossl.dylib`

I've tested and removed/re-added all of the above, and the app starts without crashing with`libqca-gnupg.dylib` and/or `libqca-logger.dylib` in the folder, but crashes when `libqca-ossl.dylib` is present. 

I'm just assisting with troubleshooting and I'm not a QGIS user myself, but it looks like it behaves as it should – the app launches into an Untitled Project, which I assume is expected behavior, with a warning message at the top of the map area ("**Authentication System:** DISABLED. Resources authenticating via the system can not be accessed"). 

Screenshot:
![Image](https://github.com/user-attachments/assets/21cb13cb-b6d2-42ee-a00f-5723a7d6f97b)

Is the app usable for my end user with the ossl dylib removed, or should they refrain from using it until it's been fixed?

**Comment by caleidoscopio** [comment]
Can someone please confirm whether or not the current version is signed and notarized? I am currently unable change Privacy and Security settings so this would be an indispensable feature.

**Comment by m-kuhn** [comment]
There is a notarized testing version downloadable at https://github.com/opengisch/qgis-notarize

This is a preview of the upcoming qgis 4 version.

There are no notarized versions of qgis 3 available (yet)

**Comment by m-kuhn** [comment]
A notarized version is available for download here https://github.com/m-kuhn/QGIS/releases/tag/final-3_44_4
